proto/spacemesh/v1/smesher_types.proto

```diff
@@ -58,86 +57,74 @@ message CoinbaseResponse {
-        FILES_STATUS_PARTIAL = 2; // Some files exist and init can be continued (and may be in progress)
-        FILES_STATUS_COMPLETE = 3; // Expected data files are available and verified
-    }
-    FilesStatus files_status = 2;
-    bool init_in_progress = 3; // True if init process is currently in progress
-    uint64 bytes_written = 4; // Bytes written to the data files (actual total files sizes)
-    string error_message = 5; // Last init process error message
-    enum ErrorType {
-        ERROR_TYPE_UNSPECIFIED = 0; // Lane's favorite imposible value
-        ERROR_TYPE_FILE_NOT_FOUND = 1; // All expected post data files not found in expected path
-        ERROR_TYPE_READ_ERROR = 2; // Failure to read from a data file
-        ERROR_TYPE_WRITE_ERROR = 3; // Failure to write to a data file
+message PoSTSetupComputeProvider {
+    uint32 id = 1; // 0, 1, 2...
+    string model = 2; // e.g. Nvidia GTX 2700
+    enum ComputeApiClass {
+        COMPUTE_API_CLASS_UNSPECIFIED = 0;
+        COMPUTE_API_CLASS_CPU = 1; // useful for testing on systems without a cuda or vulkan GPU
+        COMPUTE_API_CLASS_CUDA = 2;
+        COMPUTE_API_CLASS_VULKAN = 3;
     }
-    ErrorType error_type = 6;
-}
-
-message PostStatusResponse {
-    PostStatus status = 1;
+    ComputeApiClass compute_api = 3; // A provided compute api
+    uint64 performance = 4; // Estimated performance in hashes per second
 }
 
-message PostDataCreationProgressStreamResponse {
-    PostStatus status = 1;
+// PoST setup options, used to define the setup requirements.
+message PoSTSetupOpts {
+    string data_dir = 1; // User provided path to create the setup data files at
+    uint32 num_units = 2; // Number of PoST data units to generate
+    uint32 num_files = 3; // Number of files to equally distribute the data among
+    uint32 compute_provider_id = 4; // A `PoSTSetupComputeProvider` id
+    bool throttle = 5; // Throttle down setup phase computations while user is interactive on system
 }
 
-message EstimatedRewardsRequest {}
-
-// Estimated rewards for the next epoch. Note that this is a global value that will be the
-// same for all smeshers, and is based on the default minimum commitment size and fixed
-// epoch length, both of which are network params.
-message EstimatedRewardsResponse {
-    // The amount of the total estimated reward in the next upcoming epoch
-    Amount amount = 1;
-    // The commitment size that this estimated reward corresponds to (part of global config)
-    uint64 data_size = 2;
+message PoSTSetupStatus {
+    enum State {
+        STATE_UNSPECIFIED = 0; // Lane's favorite impossible value
+        STATE_NOT_STARTED = 1; // Setup not started
+        STATE_IN_PROGRESS = 2; // Setup in progress
+        STATE_COMPLETE = 3; // Setup is complete
+        STATE_ERROR = 4; // Setup last session ended with an error
+    }
+    State state = 1;
+    uint64 num_labels_written = 4; // Number of labels (hashes) written to the data files
```

Comment:
Why are these out of order?

Reply:
fixed: https://github.com/spacemeshos/api/pull/145/commits/3679ef8d0c03690f580ff23e6db11855cc223af8